<launch>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <arg name="drone_id" value="0"/>

    <arg name="map_size_x" value="20.0"/>
    <arg name="map_size_y" value="10.0"/>
    <arg name="map_size_z" value=" 2.0"/>
    <arg name="odom_topic" value="visual_slam/odom" />
    
    <!-- map -->
    <node pkg ="map_generator" name ="map_generator" type ="dynamic_forest_seq" output = "screen">
        <param name="init_state_x"   value="0.0"/>
        <param name="init_state_y"   value="0.0"/>
        <param name="map/x_size"     value="$(arg map_size_x)" />
        <param name="map/y_size"     value="$(arg map_size_y)" />
        <param name="map/z_size"     value="$(arg map_size_z)" />
        <param name="map/resolution" value="0.1"/>        
        <param name="map/frame_id"     value="world" />
    
        <param name="ObstacleShape/seed" value="-1"/>
        <param name="map/obs_num"    value="10"/>
        <param name="ObstacleShape/lower_rad" value="0.3"/>
        <param name="ObstacleShape/upper_rad" value="1.0"/>
        <param name="ObstacleShape/lower_hei" value="5.0"/>
        <param name="ObstacleShape/upper_hei" value="5.0"/>        
        <param name="ObstacleShape/set_cylinder" value="true"/>     
    
        <param name="map/circle_num" value="10"/>        
        <param name="ObstacleShape/radius_l" value="0.7"/>        
        <param name="ObstacleShape/radius_h" value="2.5"/>        
        <param name="ObstacleShape/z_l" value="0.8"/>        
        <param name="ObstacleShape/z_h" value="2.0"/>        
        <param name="ObstacleShape/dr" value="0.3"/>        
        <param name="ObstacleShape/theta" value="0.2"/>        
        <param name="ObstacleShape/omega" value="0.05"/>        
    
        <param name="sensing/radius" value="5.0"/>        
        <param name="sensing/rate"   value="5.0"/>   
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find map_generator)/launch/default.rviz" required="true" />
      
</launch>
